File: classes/parte-10-interoperabilidad-y-fronteras-entre-lenguajes/164-elegir-el-lenguaje-correcto-para-cada-componente/implementaciones/go/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	line, _ := bufio.NewReader(os.Stdin).ReadString('\n')
	tipo := strings.TrimSpace(line)
	rec := map[string]string{"sistemas": "Rust", "web": "TypeScript", "datos": "SQL"}
	r, ok := rec[tipo]
	if !ok {
		r = "Python"
	}
	fmt.Printf("lenguaje=%s\n", r)
}

File: classes/parte-10-interoperabilidad-y-fronteras-entre-lenguajes/164-elegir-el-lenguaje-correcto-para-cada-componente/implementaciones/java/Main.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String tipo = br.readLine().trim();
        String r;
        switch (tipo) {
            case "sistemas": r = "Rust"; break;
            case "web": r = "TypeScript"; break;
            case "datos": r = "SQL"; break;
            default: r = "Python";
        }
        System.out.println("lenguaje=" + r);
    }
}
